Develop

teodorakocan opened the pull request.

teodorakocan committed:
pocetni izgled

teodorakocan committed:
dodavanje i izlistavanje taskova

teodorakocan committed:
brisanje taskova

teodorakocan committed:
filtriranje

teodorakocan committed:
ispravka filtriranja

AirmiX commented on todo.html:
```diff
@@ -0,0 +1,335 @@
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" type="text/css" href="style.css">
+    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/4.1.0/css/bootstrap.min.css" rel="stylesheet"/>
```
Don't use bootstrap

AirmiX commented on todo.html:
```diff
@@ -0,0 +1,335 @@
+            complitedTasks = [];
+            allTasks = [];
+            localStorage.setItem("tasks", allTasks);
+            localStorage.setItem("active", activeTasks);
```
Why are you storing both, only `tasks` should be enough. You can always filter those that are active

AirmiX commented on todo.html:
```diff
@@ -0,0 +1,335 @@
+        </div>
+    </div>
+
+    <script>
```
You can do this:

```
const tasksString = localStorage.getItem("tasks");

if (!tasksString) {   // meaning tasksString is either null, undefined or '' 
    ...
} else {
    const tasks = JSON.parse(tasksString);   // tasks is an array
    ...
```

AirmiX reviewed (changes requested):
- Rename todo.html to index.html (that is the name of the main HTML page)
- Move Javascript code to separate file
- Don't use var, use const and let only (if the value doesn't change use const)
- Naming:
   - spelling: complited -> completed
   - name functions in camelcase, not pascal case (lower the first letter)
   - name variables in camelcase, not snake case ()
   - no need to use data structure name as part of variable name (`activeArray` -> `activeTasks`) 
- Store tasks like this, so you don't need to store active and completed separately:
```
[{ name: 'task 1', completed: false }, { name: 'task 2', completed: true }, ...]
```
- Bugs:
    - active filter doesn't work well
    - try adding two tasks with the same name :)
- Follow comments
- Code style: 
```
// don't
if () 
{
}
else 
{

// do
if () {

} else {

```

AirmiX commented on todo.html:
```diff
@@ -0,0 +1,335 @@
+                activeTasks = allActive;
+                complitedTasks = allComplited;
+                localStorage.setItem('active', allActive);
+                localStorage.setItem('complited', allComplited);
```
Don't store active and completed tasks, you can always filter them from tasks

teodorakocan committed:
removed the bootstrap and upgraded a custom style

teodorakocan committed:
corrections

-todo.html file renamed to index.html,
-instead of var i used let and const,
-correction of spelling,
-corrected names of functions,
-changed the way of storing tasks,
-removed bugs in filtering activ tasks and in function of adding the new task with the same name as existing task in storage
-changed code style
-

AirmiX reviewed (changes requested):
Use const where you can. An example below is not changing the reference of the variable `a` so you can use `const`.
```
const a = {};
a.b = 3;
a.c = 5
```

When you move js code to separate file, figure out the shortcut for text formatting in your text editor and format code. That will fix spacing and indentation.

Decide on using 'text' or "text", don't mix both. Let's go with single quotes 'text'.

Handle long names, the delete button goes outside the component. When the name is long it should be trimmed with `...`. This can be handled in CSS.

Clear input after adding task (pressing 'Enter')

Follow other comments

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+        input.addEventListener("keyup", function(event) {
+            if (event.keyCode === 13) {
+                event.preventDefault();
+                if(input.value != ''){
```
Use strict comparison always (`!==`). In this case you can even do just `if(input.value)`

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+    </div>
+
+    <script>
+        const tasksString = localStorage.getItem("tasks");
```
Create constant for 'tasks' and reuse it everywhere.

`const TASKS_KEY = 'tasks';`

`localStorage.getItem("tasks") -> localStorage.getItem(TASKS_KEY)`
`localStorage.setItem("tasks", ...) -> localStorage.setItem(TASKS_KEY, ...)`

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+    <title>VanillaJS &#8226; TodoMVC</title>
+</head>
+<body style="background-color:rgb(234, 238, 238)">
```
do `body { ...` in css instead

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+            let btnAll = document.getElementById("all");
+            btnAll.classList.remove("active");
+            let btnComplited = document.getElementById("completed");
+            btnComplited.classList.remove("active");
```
btnCompleted

AirmiX replied:
Fix this in every function

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+            let btnAll = document.getElementById("all");
+            btnAll.classList.remove("active");
+            let btnComplited = document.getElementById("completed");
+            btnComplited.classList.add("active");
```
btnCompleted

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+            for(let i = 0; i < allTasks.children.length; i++){
+                let myobj = document.getElementById(allTasks.children[i].id);
+                i--;
+                myobj.remove();
```
Two things:
- Use camelCase (myObj)
- Never name variables myObj, myFunction, etc.

AirmiX replied:
Fix this everywhere

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+        }
+
+        function checkFilter(){
+            let btnAll = document.getElementById("all");
+            let btnComplited = document.getElementById("completed");
+            let btnActive = document.getElementById("active");
```
These buttons could go outside the function since they don't change. Then you don't need to search for them in document on every function execution

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+
+        let numb = document.createElement('label');
+        numb.setAttribute("id", "numbOfActive")
+        numb.innerText = numOfActiveTasks + " items left";
```
Rename. Eg. numberOfTasksEl. Or txtNumberOfActiveTasks/txtTasksLeft, since you already named buttons like btnName...)

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+        }
+
+        function crossTask(value){
+            if(document.getElementById("input_" + tasks.indexOf(value)).checked){
+                tasks[tasks.indexOf(value)].completed = true;
+                localStorage.setItem("tasks", JSON.stringify(tasks));
+            }
+            else{
+                tasks[tasks.indexOf(value)].completed = false;
+                localStorage.setItem("tasks", JSON.stringify(tasks));
+            }
```
You can do like this:

```
tasks[tasks.indexOf(value)].completed = document.getElementById("input_" + tasks.indexOf(value)).checked;
localStorage.setItem("tasks", JSON.stringify(tasks));
```

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+            }
+        }
+
+        function reloadNumbOfTasks(){
```
reloadNumberOfTasks, no need to shorten it

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+        }
+
+        function reloadNumbOfTasks(){
+            let numb = document.getElementById('numbOfActive');
```
you already have this reference at the beginning of the file and it is not changing during the app lifetime

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+        }
+
+        function allTasks(){
+            let btnAll = document.getElementById("all");
+            btnAll.classList.add("active");
+            let btnComplited = document.getElementById("completed");
+            btnComplited.classList.remove("active");
+            let btnActive = document.getElementById("active");
+            btnActive.classList.remove("active");
```
extract this into function since it is repeating 3 times

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+            for(let i = 0; i < allTasks.children.length; i++)
+            {
+                let myobj = document.getElementById(allTasks.children[i].id);
+                i--;
```
al si ga haknula xD 
Use while in this case
```
while (allTasks.children.length > 0) {
    const myobj = document.getElementById(allTasks.children[i].id);
    myobj.remove();
}
```

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+            }
+
+            tasks = deleted;
+            localStorage.removeItem("tasks");
```
no need to remove

AirmiX commented on index.html:
```diff
@@ -0,0 +1,220 @@
+            tasks = deleted;
+            localStorage.removeItem("tasks");
+            localStorage.setItem("tasks", JSON.stringify(tasks));
+            location.reload();
```
don't reload the page

teodorakocan committed:
changes

- separated JavaScriptCode from html code
- changed "tasks" into 'tasks'
- handled long inputs
-after entering tasks input area will be empty for adding new task
-changed the name of some variables

AirmiX commented on index.js:
```diff
@@ -0,0 +1,177 @@
+const TASK_KEY = 'tasks';
+const index = 0;
```
Where is this used?

AirmiX commented on index.js:
```diff
@@ -0,0 +1,177 @@
+
+    tasks = deleted;
+    localStorage.setItem(TASK_KEY, JSON.stringify(tasks));
+    location.reload();
```
Don't reload the page

teodorakocan committed:
refactor after review
